ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -67,6 +87,24 @@ public static Runnable doWrap(Runnable runnable, ControlPoint controlPoint) {
         }
     }
 
+    public static <T> Callable<T> doWrapOrThrowException(Callable<T> callable, ControlPoint controlPoint) {
```

Comment:
instead of adding a new method you should fix the existent method, any usage of it has the issue